From e384e6ef0b7de485ae19b576d8332f7c685673c5 Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Thu, 18 Sep 2014 16:51:38 -0400
Subject: [powercap] intel_rapl: add new CPU IDs

Message-id: <20140918165138.28669.26864.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 96282
O-Subject: [PATCH RHEL7.1 BZ1142597 3/3] powercap / RAPL: add new CPU IDs
Bugzilla: 1142597
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1142597

Description:
------------
Add support for Broadwell model 0x3d and Haswell model (0x3c).

Signed-off-by: Jacob Pan <jacob.jun.pan@linux.intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

RHEL Version Found:
-------------------
7.0

Upstream Status:
----------------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=a97ac35b5d9e948ccfcbc04c69e9d2c94e788e94

----------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/powercap/intel_rapl.c b/drivers/powercap/intel_rapl.c
index 1bc7331..1e74ff1 100644
--- a/drivers/powercap/intel_rapl.c
+++ b/drivers/powercap/intel_rapl.c
@@ -951,7 +951,9 @@ static const struct x86_cpu_id rapl_ids[] = {
  { X86_VENDOR_INTEL, 6, 0x2d},/* Sandy Bridge EP */
  { X86_VENDOR_INTEL, 6, 0x37},/* Valleyview */
  { X86_VENDOR_INTEL, 6, 0x3a},/* Ivy Bridge */
- { X86_VENDOR_INTEL, 6, 0x45},/* Haswell */
+ { X86_VENDOR_INTEL, 6, 0x3c},/* Haswell */
+ { X86_VENDOR_INTEL, 6, 0x3d},/* Broadwell */
+ { X86_VENDOR_INTEL, 6, 0x45},/* Haswell ULT */
  /* TODO: Add more CPU IDs after testing */
  {}
 };
-- 
1.7.1